crates/bevy_ecs/src/lib.rs

```diff
@@ -2759,3 +2762,10 @@ mod tests {
         fn custom_clone(_source: &SourceComponent, _ctx: &mut ComponentCloneCtx) {}
     }
 }
+
+/// Event sent when a hotpatch happens.
+///
+/// Systems should refresh their inner pointers.
+#[cfg(feature = "hotpatching")]
+#[derive(Event, Default)]
+pub struct HotPatched;
```

Comment:
I think this location is slightly surprising as normally the test module is the last item in a file. We wouldn't normally put public types here

Reply:
well that file is 80% tests 😄